File: clients/rust/examples/balance/encryption.rs
use anyhow::Result;
use zama_sdk_sidecar::{Address, EncryptInput, EncryptOptions, EncryptParams, Sdk};

pub async fn encrypt_inputs(
    sdk: &Sdk,
    contract_address: Address,
    user_address: Address,
) -> Result<()> {
    let inputs = [
        EncryptInput::Uint64(1000.into()),
        EncryptInput::Bool(true),
        EncryptInput::Address(user_address),
    ];
    let result = sdk
        .encrypt(
            EncryptParams {
                values: &inputs,
                contract_address,
                user_address,
            },
            EncryptOptions::default(),
        )
        .await?;
    for (index, handle) in result.encrypted_values.iter().enumerate() {
        println!("Encrypted input {index}: {handle}");
    }
    println!(
        "Input proof: {}",
        alloy_primitives::hex::encode_prefixed(result.input_proof)
    );
    Ok(())
}

File: clients/go/examples/balance/encryption.go
package main

import (
	"context"
	"fmt"
	"math/big"

	"github.com/ethereum/go-ethereum/common"
	"github.com/zama-ai/sdk/clients/go/v3"
)

func encryptInputs(ctx context.Context, sdk *zama.SDKContext, contract, user common.Address) error {
	result, err := sdk.Encrypt(ctx, zama.EncryptParams{
		Values: []zama.EncryptInput{
			zama.Euint64(big.NewInt(1000)),
			zama.Ebool(true),
			zama.Eaddress(user),
		},
		ContractAddress: contract,
		UserAddress:     user,
	}, zama.EncryptOptions{})
	if err != nil {
		return err
	}
	for i, value := range result.EncryptedValues {
		fmt.Printf("Encrypted input %d: %s\n", i, value.Hex())
	}
	fmt.Printf("Input proof: 0x%x\n", result.InputProof)
	return nil
}
